modules/programs/starship.nix

```diff
@@ -30,6 +30,34 @@ in {
       description = "The package to use for the starship binary.";
     };
 
+    disabled = mkOption {
```

Comment:
I don't see what advantage these options offer over setting them in the `settings` option. Are you mainly looking for the improved merge behavior? I think the correct way forward in that case would be to improve the type of `settings`.

Reply:
Possibly. I just wanted something declarative

Reply:
I'm still not sure what idiomatic nix is

Reply:
I made a PR to introduce a more elaborate type for the `settings` option. This new type should allow correct merging of configurations (see the test case in the PR).